org-code-javabuilder/lib/src/main/java/org/code/javabuilder/PerformanceTracker.java

```diff
@@ -0,0 +1,60 @@
+package org.code.javabuilder;
+
+import static org.code.protocol.LoggerNames.MAIN_LOGGER;
+
+import java.time.Clock;
+import java.time.Instant;
+import java.util.logging.Logger;
+import org.code.protocol.LoggerConstants;
+import org.json.JSONObject;
+
+public class PerformanceTracker {
+  private static final String FIRST_INSTANCE = "firstInstance";
+  private static final String COLD_BOOT_START = "coldBootStart";
+  private static final String COLD_BOOT_END = "coldBootEnd";
+  private static final String INSTANCE_START = "instanceStart";
+  private static final String COMPILE_START = "compileStart";
+  private static final String COMPILE_END = "compileEnd";
+  private static final String USER_CODE_START = "userCodeStart";
+  private static final String USER_CODE_END = "userCodeEnd";
+  private static final String INSTANCE_END = "instanceEnd";
+  private static final String TYPE = "performanceReport";
+  private final JSONObject logs;
+
+  public PerformanceTracker() {
+    logs = new JSONObject();
+  }
+
+  public void trackStartup(
+      Instant coldBootStart, Instant coldBootEnd, Instant instanceStart, boolean firstInstance) {
+    logs.put(COLD_BOOT_START, coldBootStart.toEpochMilli());
+    logs.put(COLD_BOOT_END, coldBootEnd.toEpochMilli());
+    logs.put(INSTANCE_START, instanceStart.toEpochMilli());
+    logs.put(FIRST_INSTANCE, firstInstance);
```

Comment:
If firstInstance = false, does this technically qualify as a cold boot? From what I can well, we're reporting this metric every time the handler is invoked right?

Reply:
Adjusted so it only reports on the first usage of the container